packages/contract/lib/reason.js

```diff
@@ -28,25 +28,24 @@ const reason = {
       // be updated to respect (instead of computing here)
       const data = res.error.data;
       let resData;
-      if ("result" in data) {
-        // there is a single result (only in ganache 3.0+)
+      if (typeof data === "string") {
+        resData = data; // geth, Ganache >7.0.0
+      } else if ("result" in data) {
+        // there is a single result (Ganache 7.0.0)
         resData = data.result;
       } else {
         // handle `evm_mine`, `miner_start`, batch payloads, and ganache 2.0
         // NOTE this only works for a single failed transaction at a time.
         const hash = Object.keys(data)[0];
         const errorDetails = data[hash];
         resData =
-          errorDetails.result /* ganache 3.0 */
-          || errorDetails.return /* ganache 2.0 */;
+          errorDetails.result /* ganache 3.0 */ ||
```

Comment:
while you are doing this would you mind updating this comment to v7? - feel free to ignore this if this is annoying

Reply:
this line was only true when ganache v7 was going to be called 3.0.0. It should be safe to remove this check.

Reply:
Oh, good to know!  I'll go change that quickly.